prime-router/src/main/kotlin/azure/OpenAPIDocsFunction.kt

```diff
@@ -0,0 +1,94 @@
+package gov.cdc.prime.router.azure
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.HttpStatus
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.BindingName
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import org.apache.logging.log4j.kotlin.Logging
+import java.io.ByteArrayOutputStream
+import java.io.InputStream
+import java.net.URL
+
+const val defaultConnEnvVar = "AzureWebJobsStorage"
+const val PARAM_RESOURCE_NAME = "resourceName"
+const val textJS = "text/javascript"
+const val textCSS = "text/css"
+const val textYaml = "text/yaml"
+val RESOURCES_NAMES = listOf<String>(
+    "api.yaml",
+    "index.css",
+    "swagger-ui.css",
+    "swagger-initializer.js",
+    "swagger-ui-bundle.js",
+    "swagger-ui-standalone-preset.js",
+    "favicon-16x16.png",
+    "favicon-32x32.png",
+)
+/**
+ * Class for serving swagger UI page
+ */
+class OpenAPIDocsFunction : Logging {
```

Comment:
> Thinking about this over the weekend and I'm wondering if we might be better off just putting this in an azure blob container and turning on [static site serving](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-static-website-host). Seems like it might be a better fit and would avoid needing to keep this up to date as the swagger ui changes. For example, if the ui version included a `*.webp` file this solution would not work.

Thought about azure static site serving and also prototyped serving from azure storage blob container serving, both have the drawbacks of extra deployment work and the CORS settings on both the storage service side and azure functions side need to relaxed (which is not recommended in most situations)

Reply:
Ah I see, but yea I was more thinking it just not being available on reportstream.cdc.gov and simply being a static azure site (or an option to assign another domain at a later point).

Agreed on the extra deployment work

Reply:
API docs are usually accessible from static site, but the actual API lives on a PROD server, a common practice:

add a link to the static site e.g. top menu "API Docs" that points to the actual API server, check out the CMS blue button site https://bluebutton.cms.gov/developers/, this is static site hosting developer documents, and the "API Docs" link to API staging server at: https://sandbox.bluebutton.cms.gov/docs/openapi

![image](https://github.com/CDCgov/prime-reportstream/assets/135686833/1e2ea648-301a-4bfd-9336-f1326f4f2de7)

Reply:
Yes, but the actual assets for that site are served out of S3.  In this implementation, we have written an azure function that serves the assets for us.

This screenshot is the the network calls from the site you linked which shows the JS bundle being loaded out of an S3 bucket.
<img width="707" alt="Screenshot 2023-07-18 at 7 34 07 AM" src="https://github.com/CDCgov/prime-reportstream/assets/1225895/eccfb781-322c-4176-95ef-01637e764155">

Reply:
yea, sub-resources of swagger ui is provisioned from s3, but the swagger ui page (entry point) is served through BB2 API end point and is loaded from here:
https://github.com/CMSgov/bluebutton-web-server/tree/master/apps/docs/templates

Reply:
good observation, the openapi.html is a django template made from swagger ui index.html, and see the template variable e.g. , the "static" will instruct the runtime (template engine) to lookup statically - which is from the web root of the API server (not s3 server), at the deployment time, the artifacts in the s3 bucket is pulled into the ngix server - i.e. - the bb2 server's web root so that the secondary resources could be found co-host... 

<link rel="icon" type="image/png" href="{% static 'swagger-ui-4.15.5-dist/favicon-32x32.png' %}" sizes="32x32" />

Reply:
Maybe we chat about this as a 17th today?  I'm not seeing that behavior in the example you linked.


This is what is in the response when I fetch https://sandbox.bluebutton.cms.gov/docs/openapi which then contains resource links to assets directly in the S3 bucket that serves the files.
```html
    <body>
        <div id="swagger-ui"></div>
        <script src="https://s3.amazonaws.com/content-impl-bbonfhir-com/static/swagger-ui-4.15.5-dist/swagger-ui-bundle.js" charset="UTF-8"></script>
        <script src="https://s3.amazonaws.com/content-impl-bbonfhir-com/static/swagger-ui-4.15.5-dist/swagger-ui-standalone-preset.js" charset="UTF-8"></script>
        <script>
            window.onload = function() {
                // Begin Swagger UI call region
                const ui = SwaggerUIBundle({
                    urls: [{
                        url: "https://s3.amazonaws.com/content-impl-bbonfhir-com/static/openapi.yaml",
                        name: "openapi.yaml"
                    }, ],
                    dom_id: '#swagger-ui',
                    deepLinking: true,
                    presets: [SwaggerUIBundle.presets.apis, SwaggerUIStandalonePreset],
                    layout: "StandaloneLayout"
                });
                // End Swagger UI call region

                window.ui = ui;
            }
            ;
        </script>
    </body>
```

Reply:
sounds good - as a 17th

you're right - seems {% 'static'  uri-value  } is resolved to s3 bucket, that makes sense - since the s3 bucket is public access (no CORS configure needed), note, this is only for the secondary resources...

let's chat as 17th